platform/mbed_critical.c

```diff
@@ -23,6 +23,21 @@
 #include "platform/mbed_assert.h"
 #include "platform/mbed_toolchain.h"
 
+// if __EXCLUSIVE_ACCESS rtx macro not defined, we need to get this via own-set architecture macros
+#ifndef __EXCLUSIVE_ACCESS
+#ifndef MBED_EXCLUSIVE_ACCESS
+#if ((__ARM_ARCH_7M__      == 1U) || \
```

Comment:
i would suggest adding any kind of reminder to update this upon new architecture arrival

Reply:
Added in the new commit